src/server/worker/workerrequests.js

```diff
@@ -882,6 +882,56 @@ function WorkerRequests(mainLogger, gmeConfig, webgmeUrl) {
             context,
             storage,
             blobClient = getBlobClient(webgmeToken),
+            checkSelectionValidity = function (parent, info) {
+                var deferred = Q.defer(),
```

Comment:
It looks like you should be able to use async/await since we are testing from node 8 on. This code might be simplified a little if you used async/await (though it is your call!).

Reply:
(Async/await would allow you to remove the deferred and the `hasError` variable in favor of simply throwing an error.)

Reply:
good point, will do in the future :)